English version of the "Building a Website from Scratch with full BEM stack" blog post

rasulksirov opened the pull request:
English translation of the article

innabelaya commented on articles/bem-full-stack/bem-full-stack.en.md:
```diff
@@ -0,0 +1,1938 @@
+
+![](https://github.com/bem/bem-method/raw/bem-info-data/articles/bem-full-stack/0-intro.jpg)
+
+Last week <a href="http://www.bbc.co.uk/blogs/internet/entries/47a96d23-ae04-444e-808f-678e6809765d">the BBC wrote</a> about using the Yandex-developed BEM methodology for the new version of their home page. This provides an occasion for us to revisit our master class «<a href="https://ru.bem.info/talks/bemup-moscow-2014/#Мастер-класс-Разрабатываем-сайт-с-нуля-на-полном-стеке-БЭМ-технологий--Жека ...
```
please remove quotation marks from the link to the master class

innabelaya commented on articles/bem-full-stack/bem-full-stack.en.md:
```diff
@@ -0,0 +1,1938 @@
+
+In this article we will discuss the benefits of front-end development based on independent blocks, find out about redefinition levels, look at ready-made block libraries and build automation tools. We will see how tools like [autoprefixer](https://github.com/postcss/autoprefixer), the CSS preproccesor [Stylus](http://learnboost.github.io/stylus/) or the modular system [YModules](https://ru.bem.in ...
+
+We will use a real-life example to explain the advantages of a declarative approach, whereby the same ideas can be used both for CSS and for JavaScript. We will specifically focus on [BEMHTML](https://ru.bem.info/technology/bemhtml/) and [BEMTREE](https://ru.bem.info/technology/bemtree/) declarative templates, which convert data to a BEM tree described in [BEMJSON](https://ru.bem.info/technology/ ...
```
ru.bem.info —> en.bem.info здесь и далее по тексту

innabelaya commented on articles/bem-full-stack/bem-full-stack.en.md:
````diff
@@ -0,0 +1,1938 @@
+…
+```
+
+If you are interested in the details of how and why BEM came into being, please read Vitaly Harisov's article [The History of BEM](https://bem.info/method/history/) and watch the [videos](https://events.yandex.ru/lib/talks/95/).
````
bem.info —> en.bem.info

zxqfox replied:
the videos → videos.

rasulksirov replied:
I would stick with the "the videos" here - since the videos in question are on the same topic as Vitaly's article mentioned earlier in the sentence; so the definite article refers us to that topic, implying that's their not just any videos but ones related to BEM history.

zxqfox replied:
@rasulksirov Okey. As a shorthand to `these` it's possible to use `the` with plurals. But whether it's permissible for documentation?

rasulksirov replied:
@zxqfox Not quite sure what you mean, but if the concern is whether it's too informal in the context of for the document, I don't see any problem, the use of the article "the" has no such effect here.

zxqfox replied:
@rasulksirov Thanks for clarification :+1:

rasulksirov commented:
Thanks for the corrections - updated the doc accordingly

tadatuta commented on articles/bem-full-stack/bem-full-stack.en.md:
```diff
@@ -0,0 +1,1938 @@
+                time: Math.floor((Math.random()*12)+1) + 'h',
+                img: service === 'instagram' ? 'http://bla.jpg' : undefined,
+                text: [
+                    'Блок — это независимый интерфейсный компонент. Блок может быть простым или составным (содержать другие блоки).',
```
I think such occasions should be translated as well

rasulksirov commented:
Remaining bits of Russian (such as pointed out by tadatuta) translated.

zxqfox commented on articles/bem-full-stack/bem-full-stack.en.md:
```diff
@@ -0,0 +1,1938 @@
+
+![](https://github.com/bem/bem-method/raw/bem-info-data/articles/bem-full-stack/0-intro.jpg)
+
+Last week <a href="http://www.bbc.co.uk/blogs/internet/entries/47a96d23-ae04-444e-808f-678e6809765d">the BBC wrote</a> about using the Yandex-developed BEM methodology for the new version of their home page. This provides an occasion for us to revisit our master class <a href="https://ru.bem.info/talks/bemup-moscow-2014/#Мастер-класс-Разрабатываем-сайт-с-нуля-на-полном-стеке-БЭМ-технологий--Жека- ...
```
That's strange to have relative dates `Last week` (?) in article with jumping publish date.
Also not sure but probably better to replace tags with md.

zxqfox replied:
BEM technology or methodology? Full stack of BEM-related technologies?

rasulksirov commented:
Changed "Last week" to "Earlier this year" - should be fine, assuming that the article gets published this year :). Replaced the tags with md, as suggested, so now the link formatting is consistent throughout the document.
As for the BEM technology/methodology conundrum, I've left it as it is for now - and anyway, it's a direct translation from the Russian (полный стек) БЭМ-технологий. Even the bem.info homepage says "BEM. Technology for creating Web applications". But personally, I've always assumed the phrase BEM technology stack (as well as стек БЭМ-технологий) to imply "stack of technologies related to BEM", and that's how the reader is meant to understand it.

zxqfox commented:
@rasulksirov "Earlier this year" is still relative and should be fixed in the next year. But since I hope it will be a lot of changes we could fix it later. Relative dates are very familiar to human.

Oh, don't think that I blame you for bad translation. Just perhaps we also need to change this in russian article.

Consider me a grammar nazi ;-)

And thanks for contribution!

rasulksirov commented:
@zxqfox Thanks for your comments and suggestions, I do realize you're not blaming me, indeed any discussion or suggestions are useful and welcome, be it about the translation per se, or the text in general.

Actually, I just realized I myself forgot to reply to one of the comments above - apologies, added a line note now.

If the relative time references are not critical for now (at least, pending corresponding changes to the Russian version first) and there are no further comments on the translation, perhaps the file could be now merged.

dab commented:
@rasulksirov can you please squash commits?

rasulksirov commented:
@dab Commits squashed (apologies for some delay).

dab commented:
@rasulksirov thank you.
